engine/src/test/java/io/camunda/zeebe/engine/perf/JobBackoffRestoreMigrationPerformanceTest.java

```diff
@@ -0,0 +1,126 @@
+@Measurement(iterations = 50, time = 1)
+@Fork(
+    value = 1,
+    jvmArgs = {"-Xmx4g", "-Xms4g"})
+@BenchmarkMode(Mode.Throughput)
+@OutputTimeUnit(TimeUnit.SECONDS)
+@State(org.openjdk.jmh.annotations.Scope.Benchmark)
+public class JobBackoffRestoreMigrationPerformanceTest {
+
+  public static final Logger LOG =
+      LoggerFactory.getLogger(JobBackoffRestoreMigrationPerformanceTest.class.getName());
+
+  public MutableJobState jobState;
+  private ZeebeDb zeebeDb;
+  private MutableProcessingState processingState;
+  private TransactionContext transactionContext;
+  private long count;
+
+  @Benchmark
+  public long measureMigrationTime(final JobBackoffRestoreMigrationPerformanceTest state) {
+    final var startTime = System.currentTimeMillis();
+    state.jobState.restoreBackoff();
+    final var stopTime = System.currentTimeMillis();
+    return stopTime - startTime;
+  }
+
+  @JMHTest("measureMigrationTime")
+  void shouldProcessWithinExpectedDeviation(final JMHTestCase testCase) {
+    // given - an expected ops/s score, as measured in CI
+    // when running this test locally, you're likely to have a different score
+    final var referenceScore = 450;
+    setupState();
+
+    // when
+    final var assertResult = testCase.run();
+
+    // then
+    assertResult.isWithinDeviation(referenceScore, 0.15);
+  }
+
+  private void setupState() {
+    jobState = processingState.getJobState();
+    final DbLong jobKey = new DbLong();
+    final DbForeignKey<DbLong> fkJob = new DbForeignKey<>(jobKey, ZbColumnFamilies.JOBS);
+    final JobStateValue jobStateValue = new JobStateValue();
+    final JobRecordValue jobRecordToRead = new JobRecordValue();
+    final ColumnFamily<DbLong, JobRecordValue> jobsColumnFamily =
+        zeebeDb.createColumnFamily(
+            ZbColumnFamilies.JOBS, transactionContext, jobKey, jobRecordToRead);
+
+    final ColumnFamily<DbForeignKey<DbLong>, JobStateValue> statesJobColumnFamily =
+        zeebeDb.createColumnFamily(
+            ZbColumnFamilies.JOB_STATES, transactionContext, fkJob, jobStateValue);
+
+    final DbLong backoffKey = new DbLong();
+    final DbCompositeKey<DbLong, DbForeignKey<DbLong>> backoffJobKey =
+        new DbCompositeKey<>(backoffKey, fkJob);
+    final ColumnFamily<DbCompositeKey<DbLong, DbForeignKey<DbLong>>, DbNil> backoffColumnFamily =
+        zeebeDb.createColumnFamily(
+            ZbColumnFamilies.JOB_BACKOFF, transactionContext, backoffJobKey, DbNil.INSTANCE);
+
+    final int maxJobCount = 200_000;
+
+    LOG.info("Creating {} jobs, please hold the line...", maxJobCount);
+    for (int i = 0; i < maxJobCount; i++) {
+      final JobRecord record = createJobRecord(1000);
+      jobKey.wrapLong(i + 1);
+      jobState.create(jobKey.getValue(), record);
+      jobState.fail(jobKey.getValue(), record);
+      backoffKey.wrapLong(record.getRecurringTime());
+      backoffColumnFamily.deleteExisting(backoffJobKey);
+      count++;
+      RecordingExporter.reset();
```

Comment:
I don't think you need to reset the exporter here. You are not writing any records

Reply:
Yeah, is not the final test let me say. But the problem is that the test is not working, I'm able to create the state but then I got a NPE